nixos/modules/services/web-apps/mastodon.nix

```diff
@@ -575,6 +583,38 @@ in {
       path = with pkgs; [ file imagemagick ffmpeg ];
     };
 
+
+  systemd.services.mastodon-media-prune =
+    let
+      cfg = config.services.mastodon;
+    in
+    lib.mkIf cfg.mediaAutoRemove {
+      description = "Mastodon media prune";
+      environment = lib.filterAttrs (n: _: n != "PATH") config.systemd.services.mastodon-web.environment;
+      serviceConfig = {
+        Type = "oneshot";
+        # Remove remote media attachments and preview cards older than one month.
+        ExecStart = ''
+          ${cfg.package}/bin/tootctl media remove --days=30
+          ${cfg.package}/bin/tootctl preview_cards remove --days=30
+        '';
```

Comment:
Does this actually work? If not, you should probably use `script` instead of `serviceConfig.ExecStart`.